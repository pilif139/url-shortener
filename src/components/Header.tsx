"use client"

import Link from "next/link";
import {motion} from "framer-motion";

export default function Header() {
  return (
      <motion.header
          initial={{opacity: 0, y: -50}}
          animate={{opacity: 1, y: 0}}
          transition={{duration: 0.5}}
          className="absolute top-0 w-full flex justify-center md:p-10">
        <Link href={"/"}>
          <h1 className="md:text-6xl text-5xl hover:bg-slate-300 transition p-5 rounded-full">
            <span className="text-violet-500 hover:text-violet-800 transition">URL</span>
            &nbsp;Shortener
          </h1>
        </Link>
      </motion.header>
  );
}