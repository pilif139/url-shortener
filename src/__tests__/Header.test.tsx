import Header from "@/components/Header";
import {render, screen} from "@testing-library/react";
import {jest} from "@jest/globals"

describe("Header Component", ()=>{
    it("should render correctly", ()=>{
        render(<Header/>);
        expect(screen.getByRole("banner")).toBeInTheDocument();
    })

    it("should render correctly header", () => {
        const {container} = render(<Header />);
        expect(container).toMatchSnapshot();
    });

    it("should have a link to the home page", () => {
        render(<Header />);
        expect(screen.getByRole("link")).toHaveAttribute("href", "/");
    });

    it("should reload the page when the link is clicked and the pathname is '/'", () => {
        Object.defineProperty(window, 'location', {
            writable: true,
            value: { pathname: "/", reload: jest.fn() }
        });
        render(<Header />);
        screen.getByRole("link").click();
        expect(window.location.reload).toHaveBeenCalled();
    });


    it("should not reload the page when the link is clicked and the pathname is not '/'", () => {
        Object.defineProperty(window, 'location', {
            writable: true,
            value: { pathname: "/", reload: jest.fn() }
        });
        window.location = { pathname: "/other", reload: jest.fn() } as any;
        render(<Header />);
        screen.getByRole("link").click();
        expect(window.location.reload).not.toHaveBeenCalled();
    });
})