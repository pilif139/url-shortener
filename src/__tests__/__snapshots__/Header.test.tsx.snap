// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly header 1`] = `
<div>
  <header
    class="flex w-full justify-center md:p-10"
    style="opacity: 0; will-change: opacity,transform; transform: translateY(-50px);"
  >
    <a
      href="/"
    >
      <h1
        class="rounded-full p-5 text-5xl transition duration-500 hover:bg-slate-300 md:text-6xl dark:hover:bg-slate-700"
      >
        <span
          class="text-violet-500 transition hover:text-violet-700 dark:hover:text-violet-400"
        >
          URL
        </span>
        Â Shortener
      </h1>
    </a>
  </header>
</div>
`;
